packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -334,9 +334,11 @@ export const useField = <
 
   useEnhancedEffect(() => {
     if (selectedSectionIndexes == null) {
-      if (inputRef.current!.selectionStart !== 0 || inputRef.current!.selectionEnd !== 0) {
-        // Ensure input selection range is in sync with component selection indexes
-        inputRef.current!.setSelectionRange(0, 0);
+      if (inputRef.current!.scrollLeft) {
+        // Ensure that input content is not marked as selected.
+        // setting selection range to 0 causes issues in Safari.
+        // https://bugs.webkit.org/show_bug.cgi?id=224425
+        inputRef.current!.scrollLeft = 0;
```

Comment:
**Off-topic**

This makes me wonder about overflows 😁

https://user-images.githubusercontent.com/3165635/226722056-795c74a2-3eda-4a31-8c9d-ace276aaf50b.mov

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

vs. 

https://user-images.githubusercontent.com/3165635/226722396-f8cc0bab-c44f-44e9-a352-a62d0bda9b40.mov

http://mui.com/x/react-date-pickers/date-time-range-field/#uncontrolled-vs-controlled